bookstore-rust/Dockerfile

```diff
@@ -0,0 +1,55 @@
+# ---------------------------------------------------
+# 1 - Build Stage
+#
+# Use official rust image to for application build
+# ---------------------------------------------------
+FROM rust:1.68.2 as build
+
+# Setup working directory
+WORKDIR /usr/src/bookstore
+COPY . .
+
+# Install dependency (Required by diesel)
+RUN apt-get update && apt-get install libpq5 -y
+
+# Build application
+RUN cargo install --path .
+
+# ---------------------------------------------------
+# 2 - Deploy Stage
+#
+# Use a distroless image for minimal container size
+# - Copy `libpq` dependencies into the image (Required by diesel)
+# - Copy application files into the image
+# ---------------------------------------------------
+FROM gcr.io/distroless/cc-debian11
+
+# Set the architecture argument (arm64, i.e. aarch64 as default)
+# For amd64, i.e. x86_64, you can append a flag when invoking the build `... --build-arg "ARCH=x86_64"`
+ARG ARCH=aarch64
+
+# libpq related (required by diesel)
+COPY --from=build /usr/lib/${ARCH}-linux-gnu/libpq.so* /usr/lib/${ARCH}-linux-gnu/
```

Comment:
is this mess auto generated or did you have to find out on your own? Does it have to be this bad?

Reply:
I tried approx 15 different docker file things ... as the OS in the container needs to provide some shared libs. So I wildy copy & pasted from the internet until one way worked ...

Probably some of the libs are not required but that would just slim down the container a couple of MBs ... it is approx 60 MB so I don't worry that much.

Reply:
It's just super ugly and the question was more out of curiousity.